packages/app/src/components/DBTraceWaterfallChart.tsx

```diff
@@ -22,233 +27,338 @@ type SpanRow = {
+}
+
+function getTableBody(tableModel: TSource) {
+  if (tableModel?.kind === SourceKind.Trace) {
+    return getSpanEventBody(tableModel) ?? '';
+  } else if (tableModel?.kind === SourceKind.Log) {
+    return tableModel.implicitColumnExpression ?? '';
+  } else {
+    return '';
+  }
+}
+
+function getFetchConfig(source: TSource, traceId: string) {
+  const alias = {
+    Body: getTableBody(source),
+    Timestamp: getDisplayedTimestampValueExpression(source),
+    Duration: source.durationExpression
+      ? getDurationSecondsExpression(source)
+      : '',
+    TraceId: source.traceIdExpression ?? '',
+    SpanId: source.spanIdExpression ?? '',
+    ParentSpanId: source.parentSpanIdExpression ?? '',
+    StatusCode: source.statusCodeExpression ?? '',
+    ServiceName: source.serviceNameExpression ?? '',
+    SeverityText: source.severityTextExpression ?? '',
+  };
+  let selectOption: {
+    valueExpression: string;
+    alias: string;
+  }[] = [];
+
+  if (source.kind === SourceKind.Trace) {
+    selectOption = [
+      {
+        valueExpression: alias.Body,
+        alias: 'Body',
+      },
+      {
+        valueExpression: alias.Timestamp,
+        alias: 'Timestamp',
+      },
+      {
+        // in Seconds, f64 holds ns precision for durations up to ~3 months
+        valueExpression: alias.Duration,
+        alias: 'Duration',
+      },
+      {
+        valueExpression: alias.SpanId,
+        alias: 'SpanId',
+      },
+      {
+        valueExpression: alias.ParentSpanId,
+        alias: 'ParentSpanId',
+      },
+      ...(alias.StatusCode
+        ? [
+            {
+              valueExpression: alias.StatusCode,
+              alias: 'StatusCode',
+            },
+          ]
+        : []),
+      ...(alias.ServiceName
+        ? [
+            {
+              valueExpression: alias.ServiceName,
+              alias: 'ServiceName',
+            },
+          ]
+        : []),
+    ];
+  } else if (source.kind === SourceKind.Log) {
+    selectOption = [
+      {
+        valueExpression: alias.Body,
+        alias: 'Body',
+      },
+      {
+        valueExpression: alias.Timestamp,
+        alias: 'Timestamp',
+      },
+      {
+        valueExpression: alias.SpanId,
+        alias: 'SpanId',
+      },
+      ...(alias.SeverityText
+        ? [
+            {
+              valueExpression: alias.SeverityText,
+              alias: 'SeverityText',
+            },
+          ]
+        : []),
+      ...(alias.ServiceName
+        ? [
+            {
+              valueExpression: alias.ServiceName,
+              alias: 'ServiceName',
+            },
+          ]
+        : []),
+    ];
+  }
+  const config = {
+    select: selectOption,
+    from: source.from,
+    timestampValueExpression: alias.Timestamp,
+    where: `${alias.TraceId} = '${traceId}'`,
+    limit: { limit: 10000 },
+    connection: source.connection,
+  };
+  return { config, alias, type: source.kind };
+}
+
+function useFetchingData({
```

Comment:
nit: I think this name is too non-descriptive, additionally reads awkwardly (I think hook names are usually `use${Noun}` format) maybe `useEventsData` or something. Same goes for the config function above

Reply:
sure thing!